tests/main.cpp

```diff
@@ -0,0 +1,69 @@
+extern "C" {
+#include <CUnit/Basic.h>
+}
+
+#include "nodes-test.h"
+#include "roles-test.h"
+
+#define ADD_TEST(__SUITE__, __MESSAGE__, __FUNC__)             \
+  if (CU_add_test(__SUITE__, __MESSAGE__, __FUNC__) == NULL) { \
+    CU_cleanup_registry();                                     \
+    return CU_get_error();                                     \
+  }
+
+int main() {
+  /* initialize the CUnit test registry */
+  if (CUE_SUCCESS != CU_initialize_registry()) {
+    return CU_get_error();
+  }
+
+  /* add a suite to the registry */
+  CU_pSuite pSuite = CU_add_suite("Suite_1", NULL, NULL);
+  if (pSuite == NULL) {
+    CU_cleanup_registry();
+    return CU_get_error();
+  }
+
+  /* add the tests to the suite */
+  /* NOTE - ORDER IS IMPORTANT */
```

Comment:
Why? Adding this kind of warning without an explanation is of little help for any future programmer :)